client.go

```
@@ -51,44 +51,35 @@ func (cli *Client) GetValue(ctx context.Context, key string) (string, error) {
 	return val, nil
 }
 
-// GetKeyValuePairs retrieves a set of key-value pairs from Redis based on a match pattern
-func (cli *Client) GetKeyValuePairs(ctx context.Context, matchPattern string, count int64) (map[string]string, error) {
-	keyValuePairs := make(map[string]string)
-	cursor := uint64(0)
-
-	// Scan through the keys in Redis
-	for {
-		// Get the list of keys matching the pattern
-		keys, newCursor, err := cli.redisClient.Scan(ctx, cursor, matchPattern, count).Result()
+// GetKeyValuePairs retrieves a set of key-value pairs from Redis based on a match pattern and a given cursor.
+func (cli *Client) GetKeyValuePairs(ctx context.Context, matchPattern string, count int64, cursor uint64) (keyValuePairs map[string]string, newCursor uint64, err error) {
+	keyValuePairs = make(map[string]string)
+
+	// Get the list of keys matching the pattern, starting from the provided cursor
+	keys, newCursor, err := cli.redisClient.Scan(ctx, cursor, matchPattern, count).Result()
+	if err != nil {
+		return nil, 0, fmt.Errorf("error scanning keys: %w", err)
+	}
+
+	// If we have keys, get the values for those keys
+	if len(keys) > 0 {
```

Should we return an error if no keys?